src/parsing.jl

```diff
@@ -44,6 +44,10 @@ function parse_source!(mod_exprs_sigs::ModuleExprsSigs, src::AbstractString, fil
         ln = count(isequal('\n'), SubString(src, 1, min(pos, length(src)))) + 1
         throw(LoadError(filename, ln, ex.args[1]))
     end
+    return process_source!(mod_exprs_sigs, ex, filename, mod; kwargs...)
+end
+
+function process_source!(mod_exprs_sigs::ModuleExprsSigs, @nospecialize(ex), filename, mod::Module; mode::Symbol=:sigs)
```

Comment:
Why is this `@nospecialize(ex)`? `ExprSplitter` has to take an `Expr` as its second argument, and calling `ExprSplitter` is the very first thing you do with `ex`.

Reply:
Here too, please don't mark as resolved if you don't answer the query.

Your docs only describe overriding the call on line 22. If you are calling this, you might need to expand your docs: what `ExprSplitter` method do people need to create? Any other methods needed? Conversely, if they don't need to override `ExprSplitter`, then that suggests `ex` is an `Expr`, and changing this to `@nospecialize(ex)` hurts the quality of type inference and makes these methods more vulnerable to invalidation.

Reply:
They don't need to override `ExprSplitter`. Is putting `ex = # julia Expr returned from running transpiler` sufficient?